tac/agents/participant/base/negotiation_behaviours.py

```diff
@@ -191,59 +199,66 @@ def on_decline(self, envelope: Envelope, dialogue: Dialogue) -> None:
         """
-        accept = envelope.message
-        assert envelope.protocol_id == "fipa" \
-            and accept.get("performative") == FIPAMessage.Performative.ACCEPT \
-            and dialogue.dialogue_label in self.game_instance.transaction_manager.pending_proposals \
-            and accept.get("target") in self.game_instance.transaction_manager.pending_proposals[dialogue.dialogue_label]
+        assert accept.get("performative") == FIPAMessage.Performative.ACCEPT \
+               and dialogue.dialogue_label in self.game_instance.transaction_manager.pending_proposals \
+               and accept.get("target") in self.game_instance.transaction_manager.pending_proposals[dialogue.dialogue_label]
         logger.debug("[{}]: on_accept: msg_id={}, dialogue_id={}, origin={}, target={}"
-                     .format(self.agent_name, accept.get("id"), accept.get("dialogue_id"), envelope.sender, accept.get("target")))
+                     .format(self.agent_name, accept.get("id"), accept.get("dialogue_id"), dialogue.dialogue_label.dialogue_opponent_pbk, accept.get("target")))
         new_msg_id = accept.get("id") + 1
         results = []
         transaction = self.game_instance.transaction_manager.pop_pending_proposal(dialogue.dialogue_label, accept.get("target"))
-        is_profitable_transaction, message = self.game_instance.is_profitable_transaction(transaction, dialogue)
-        logger.debug(message)
+        is_profitable_transaction, accept_log_msg = self.game_instance.is_profitable_transaction(transaction, dialogue)
+        logger.debug(accept_log_msg)
         if is_profitable_transaction:
             if self.game_instance.strategy.is_world_modeling:
                 self.game_instance.world_state.update_on_initial_accept(transaction)
             logger.debug("[{}]: Locking the current state (as {}).".format(self.agent_name, dialogue.role))
             self.game_instance.transaction_manager.add_locked_tx(transaction, as_seller=dialogue.is_seller)
-            results.append(Envelope(to=self.game_instance.controller_pbk, sender=self.crypto.public_key, protocol_id=ByteMessage.protocol_id,
-                                    message=ByteMessage(message_id=STARTING_MESSAGE_ID, dialogue_id=accept.get("dialogue_id"), content=transaction.serialize())))
-            results.append(Envelope(to=envelope.sender, sender=self.crypto.public_key, protocol_id=FIPAMessage.protocol_id,
-                                    message=FIPAMessage(message_id=new_msg_id, dialogue_id=accept.get("dialogue_id"), target=accept.get("id"), performative=FIPAMessage.Performative.MATCH_ACCEPT)))
+
+            msg = SimpleMessage(type=SimpleMessage.Type.BYTES, content=transaction.serialize())
+            msg_bytes = SimpleSerializer().encode(msg)
+            results.append(Envelope(to=self.game_instance.controller_pbk, sender=self.crypto.public_key, protocol_id=SimpleMessage.protocol_id, message=msg_bytes))
+            dialogue.outgoing_extend([msg])
```

Comment:
need to check if the old implementation expects this to be appended to the dialogue